subworkflows/local/mirna_quant.nf

```diff
@@ -8,121 +8,132 @@ include {   PARSE_FASTA_MIRNA  as PARSE_MATURE
 
-    PARSE_HAIRPIN ( hairpin ).parsed_fasta.set { hairpin_parsed }
+    PARSE_HAIRPIN ( ch_reference_hairpin, ch_parse_species_input )
+    ch_hairpin_parsed = PARSE_HAIRPIN.out.parsed_fasta
     ch_versions = ch_versions.mix(PARSE_HAIRPIN.out.versions)
 
-    FORMAT_HAIRPIN ( hairpin_parsed )
+    FORMAT_HAIRPIN ( ch_hairpin_parsed )
     ch_versions = ch_versions.mix(FORMAT_HAIRPIN.out.versions)
 
-    INDEX_HAIRPIN ( FORMAT_HAIRPIN.out.formatted_fasta ).index.set { hairpin_bowtie }
+    INDEX_HAIRPIN ( FORMAT_HAIRPIN.out.formatted_fasta )
+    hairpin_bowtie = INDEX_HAIRPIN.out.index.first()
     ch_versions = ch_versions.mix(INDEX_HAIRPIN.out.versions)
 
-    BOWTIE_MAP_HAIRPIN ( reads_hairpin, hairpin_bowtie.collect() )
+    BOWTIE_MAP_HAIRPIN ( ch_reads_hairpin, hairpin_bowtie, true )
     ch_versions = ch_versions.mix(BOWTIE_MAP_HAIRPIN.out.versions)
 
     BAM_STATS_HAIRPIN ( BOWTIE_MAP_HAIRPIN.out.bam, FORMAT_HAIRPIN.out.formatted_fasta )
     ch_versions = ch_versions.mix(BAM_STATS_HAIRPIN.out.versions)
 
-    BAM_STATS_MATURE.out.idxstats.collect{it[1]}
+    ch_edger_input = BAM_STATS_MATURE.out.idxstats.collect{it[1]}
         .mix(BAM_STATS_HAIRPIN.out.idxstats.collect{it[1]})
-        .dump(tag:'edger')
         .flatten()
         .collect()
-        .set { edger_input }
 
-    EDGER_QC ( edger_input )
+    EDGER_QC ( ch_edger_input )
     ch_versions.mix(EDGER_QC.out.versions)
 
-    reads
-        .map { add_suffix(it, "seqcluster") }
-        .dump (tag:'ssux')
-        .set { reads_seqcluster }
+    SEQCLUSTER_COLLAPSE ( ch_reads_for_mirna )
+    ch_versions = ch_versions.mix(SEQCLUSTER_COLLAPSE.out.versions)
 
-    SEQCLUSTER_SEQUENCES ( reads_seqcluster ).collapsed.set { reads_collapsed }
-    ch_versions = ch_versions.mix(SEQCLUSTER_SEQUENCES.out.versions)
+    ch_reads_collapsed = SEQCLUSTER_COLLAPSE.out.fastq
 
-    BOWTIE_MAP_SEQCLUSTER ( reads_collapsed, hairpin_bowtie.collect() )
+    BOWTIE_MAP_SEQCLUSTER ( ch_reads_collapsed, hairpin_bowtie, true )
     ch_versions = ch_versions.mix(BOWTIE_MAP_SEQCLUSTER.out.versions)
 
     ch_mirtop_logs = Channel.empty()
-    if (params.mirtrace_species){
-        MIRTOP_QUANT ( BOWTIE_MAP_SEQCLUSTER.out.bam.collect{it[1]}, FORMAT_HAIRPIN.out.formatted_fasta.collect{it[1]}, gtf )
-        ch_mirtop_logs = MIRTOP_QUANT.out.logs
-        ch_versions = ch_versions.mix(MIRTOP_QUANT.out.versions)
-
-        TABLE_MERGE ( MIRTOP_QUANT.out.mirtop_table )
-        ch_versions = ch_versions.mix(TABLE_MERGE.out.versions)
-    }
-    BOWTIE_MAP_HAIRPIN.out.unmapped
+
+    // nf-core/mirtop
+
+    ch_mirna_gtf_species = ch_mirna_gtf.map{ meta,gtf -> gtf }
+            .combine(ch_mirtrace_species)
+            .map{ gtf, species -> [ [id:species.toString()], gtf, species ] }
+            .collect()
+
+    BAM_STATS_MIRNA_MIRTOP(
+            BOWTIE_MAP_SEQCLUSTER.out.bam,
+            FORMAT_HAIRPIN.out.formatted_fasta,
+            ch_mirna_gtf_species )
```

Comment:
```suggestion
    BAM_STATS_MIRNA_MIRTOP(BOWTIE_MAP_SEQCLUSTER.out.bam, FORMAT_HAIRPIN.out.formatted_fasta, ch_mirna_gtf_species )
```

Pretty much everywhere else you've been one-lining the module calls and this isn't super long when collpased to one